geo/src/algorithm/line_measures/frechet_distance.rs

```diff
@@ -52,47 +52,54 @@ where
     MetricSpace: Distance<F, Point<F>, Point<F>>,
 {
     fn frechet_distance(&self, ls_1: &LineString<F>, ls_2: &LineString<F>) -> F {
-        if ls_1.coords_count() != 0 && ls_2.coords_count() != 0 {
-            Data {
-                cache: vec![F::zero(); ls_1.coords_count() * ls_2.coords_count()],
-                ls_a: ls_1,
-                ls_b: ls_2,
-            }
-            .compute_linear(self)
+        if ls_1.coords_count() == 0 || ls_2.coords_count() == 0 {
+            return F::zero();
+        }
+
+        let (ls_short, ls_long) = if ls_1.coords_count() <= ls_2.coords_count() {
+            (ls_1, ls_2)
         } else {
-            F::zero()
+            (ls_2, ls_1)
+        };
+
+        Data {
+            cache: vec![F::zero(); ls_short.coords_count() + ls_short.coords_count()],
```

Comment:
Maybe this, so it doesn't look like a type for `ls_long`?

```suggestion
            cache: vec![F::zero(); ls_short.coords_count() * 2],
```